cli/scancommands.go

```diff
@@ -398,7 +414,22 @@ func AuditSpecificCmd(c *components.Context, technology coreutils.Technology) er
 	}
 	technologies := []string{string(technology)}
 	auditCmd.SetTechnologies(technologies)
-	return progressbar.ExecWithProgress(auditCmd)
+	err = progressbar.ExecWithProgress(auditCmd)
+
+	// Reporting error to Coralogix if Xsc service is enabled
+	if err != nil && usage.ShouldReportUsage() {
+		var serverDetails *coreConfig.ServerDetails
+		serverDetails, innerError := auditCmd.ServerDetails()
+		if innerError != nil {
+			log.Error(fmt.Sprintf("failed to get server details: %q", innerError))
+			return err
+		}
+		if coralogixReportError := utils.ReportToCoralogix(serverDetails, err, "cli"); coralogixReportError != nil {
+			log.Error("failed to send error log to Coralogix:" + coralogixReportError.Error())
+		}
+	}
+
+	return err
```

Comment:
```suggestion
    func ExecCmdWithAnalytics(auditCmd *audit.AuditCommand) (err error) {
        if err = progressbar.ExecWithProgress(auditCmd); err == nil || !usage.ShouldReportUsage() {
            return
        }
        // Report error log to Analytics
        var serverDetails *coreConfig.ServerDetails
        if serverDetails, innerError := auditCmd.ServerDetails(); innerError != nil {
            log.Error(fmt.Sprintf("failed to get server details: %q", innerError))
            return
        }
        if coralogixReportError := utils.ReportToCoralogix(serverDetails, err, "cli"); coralogixReportError != nil {
			log.Error("failed to send error log to JFrog analytics:" + coralogixReportError.Error())
		}
        return
    }
```

* I would extract it to a separeted method, you have code duplication.
* You don't need to mention `Coralogix`, it doesn't matter to the user and if we will change in the future it will not be up to date.